File: dna_to_protein.R
# Translate DNA - Protein

genetic_code <- function(codon) {
  if (codon == "ATG") 
    {return("M")}
  else if (codon == "GGT" | codon == "GGC" | codon == "GGA" | codon == "GGG") 
    {return("G")}
  else if (codon == "TCT" | codon == "TCC" | codon == "TCA" | codon == "TCG" | codon == "AGT" | codon == "AGG" | codon == "AGC")
    {return("S")}
  else if (codon == "CCT" | codon == "CCC" | codon == "CCA" | codon == "CCG")
    {return("P")}
  else if (codon == "GCT" | codon == "GCC" | codon == "GCA" | codon == "GCG")
    {return("A")}
  else if (codon == "ACT" | codon == "ACC" | codon == "ACA" | codon == "ACG")
    return("T")
  else if (codon == "TAT" | codon == "TAC")
    {return("Y")}
  else if (codon == "TTT" | codon == "TTC")
    {return("F")}
  else if (codon == "TTA" | codon == "TTG" | codon == "CTT" | codon == "CTC" | codon == "CTA" | codon == "CTG")
    {return("L")}
  else if (codon == "ATT" | codon == "ATC" | codon == "ATA")
    {return("I")}
  else if (codon == "GTT" | codon == "GTC" | codon == "GTA" | codon == "GTG")
    {return("V")}
  else if (codon == "CAT" | codon == "CAC")
    {return("H")}
  else if (codon == "CAA" | codon == "CAG")
    {return("Q")}
  else if (codon == "AAT" | codon == "AAC")
    {return("N")}
  else if (codon == "AAA" | codon == "AAG")
    {return("K")}
  else if (codon == "GAT" | codon == "GAC")
    {return("D")}
  else if (codon == "GAA" | codon == "GAG")
    {return("E")}
  else if (codon == "TGT" | codon == "TGC")
    {return("C")}
  else if (codon == "TGG")
    {return("W")}
  else if (codon == "CGT" | codon == "CGC" | codon == "CGA" | codon == "CGG" | codon == "AGA" | codon == "AGG")
    {return("R")}
  else if (codon == "TAA" | codon == "TAG" | codon == "TGA")
    {return("-")}
  return()
}


dna_to_codons <- function(fasta) {
  fasta_split <- strsplit(fasta, "")[[1]]  # Lo convertimos en vect| de caracteres
  codon <- c()
  codon_list <- c()
  for (i in fasta_split) {
    codon <- c(codon, i)
    if (length(codon) == 3) {
      codon <- paste(codon[1], codon[2], codon[3], sep = "")
      codon_list <- c(codon_list, codon)
      codon <- c()
    }
  }
  return(codon_list)
}


main <- function() {
  fasta <- readline(prompt = "Introduce DNA sequence (no spaces allowed): \n")
  fasta <- gsub(" ", "", fasta)
  codon_list <- dna_to_codons(fasta)
  aa_seq <- c()
  for (i in codon_list) {
    aa_seq <- c(aa_seq, genetic_code(i))
  }
  aa_seq <- paste(aa_seq, collapse = "")
  return(aa_seq)
}


  
main()

File: dna_to_protein.py
# Translate DNA - Protein


genetic_code = {"ATG": "M",
                "GGT": "G", "GGC": "G", "GGA": "G", "GGG": "G",
                "TCT": "S", "TCC": "S", "TCA": "S", "TCG": "S", "AGT": "S", "AGC": "S",
                "CCT": "P", "CCC": "P", "CCA": "P", "CCG": "P",
                "GCT": "A", "GCC": "A", "GCA": "A", "GCG": "A",
                "ACT": "T", "ACC": "T", "ACA": "T", "ACG": "T",
                "TAT": "Y", "TAC": "Y",
                "TTT": "F", "TTC": "F",
                "TTA": "L", "TTG": "L", "CTT": "L", "CTC": "L", "CTA": "L", "CTG": "L",
                "ATT": "I", "ATC": "I", "ATA": "I",
                "GTT": "V", "GTC": "V", "GTA": "V", "GTG": "V",
                "CAT": "H", "CAC": "H",
                "CAA": "Q", "CAG": "Q",
                "AAT": "N", "AAC": "N",
                "AAA": "K", "AAG": "K",
                "GAT": "D", "GAC": "D",
                "GAA": "E", "GAG": "E",
                "TGT": "C", "TGC": "C",
                "TGG": "W",
                "CGT": "R", "CGC": "R", "CGA": "R", "CGG": "R", "AGA": "R", "AGG": "R",
                "TAA": "-", "TAG": "-", "TGA": "-"}


def translation(codon):
    if codon in genetic_code:
        return genetic_code[codon]
    else:
        print("something is not working")
    return


def dna_to_codons(fasta):
    codon = []
    codon_list = []
    for i in fasta:
        codon.append(i)
        if len(codon) == 3:
            codon = "".join(codon)
            codon_list.append(codon)
            codon = []
    return codon_list


def main():
    fasta = input("Introduce DNA sequence: \n")
    fasta = fasta.replace(" ", "")
    codon_list = dna_to_codons(fasta)
    aa_seq = []
    for i in codon_list:
        aa_seq.append(translation(i))
    aa_seq = "".join(aa_seq)
    print("\n" + aa_seq)
    return


main()
